bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,172 @@
+
+        annot_dict['comment']["singleline"].update(
+                    lang_dict["comment_delimiter"])
+
+        annot_dict['comment']["multiline"].update(
+                    lang_dict["multiline_comment_delimiters"])
+
+        annot_dict['string']['singleline'].update(
+                                            lang_dict["string_delimiters"])
+        annot_dict['string']['multiline'].update(
+                                       lang_dict["multiline_string_delimiters"])
+        multi_str = single_str = multi_comm = ()
+        search_dict = annot_dict['string']['multiline']
+        multi_str = find_string_comment(file, search_dict, True)
+        search_dict = annot_dict['string']['singleline']
+        single_str = find_string_comment(file, search_dict, True)
+        search_dict = annot_dict['comment']["multiline"]
+        multi_comm = find_string_comment(file, search_dict, False)
+
+        match_pos += single_str + multi_str + multi_comm
+        # Seperate since single-line comments don't have a definite start and
+        # end
+        search_dict = annot_dict['comment']["singleline"]
+        single_comm = find_singleline_comments(file, search_dict, match_pos)
+        match_pos += single_comm
+        match_pos = remove_nested(match_pos)
+        m_comm = s_comm = m_str = s_str = []
+        if match_pos:
+            m_comm = list(get_range(file, filename, match_pos, multi_comm))
+            s_comm = list(get_range(file, filename, match_pos, single_comm))
+            m_str = list(get_range(file, filename, match_pos, multi_str))
+            s_str = list(get_range(file, filename, match_pos, single_str))
+        yield HiddenResult(self, m_comm)
+        yield HiddenResult(self, s_comm)
+        yield HiddenResult(self, m_str)
+        yield HiddenResult(self, s_str)
+
+
+def get_range(file, filename, match_pos, string_comment):
+    """
+    Checks if range is valid and then yields it.
+    """
+    search_range = []
+    for i in string_comment:
+        if i in match_pos:
+            search_range.append(i)
+    search_range = [(calc_line_col(file, start), calc_line_col(file, end))
+                    for (start, end) in search_range]
+
+    for i in search_range:
+        yield SourceRange.from_values(filename,
+                                      start_line=i[0][0],
+                                      start_column=i[0][1],
+                                      end_line=i[1][0],
+                                      end_column=i[1][1])
+
+
+def find_string_comment(file, annot, escape):
+    """
+    gives all instances of strings or multiline comments found within
+    the file, even if they are nested in other strings or comments
+    """
+    if escape:
+        search_func = unescaped_search_in_between
+    else:
+        search_func = search_in_between
+    found_pos = ()
+    for annot_type in annot:
+        found_pos += tuple(search_func(annot_type, annot[annot_type], file))
```

Comment:
@AbdealiJK this tuple would just have generator objects, although in my implementation those are "generated" inside the tuple.